File: web/src/types/project.ts
// 项目类别枚举
export type ProjectCategory =
  | 'SMART_BUILDING'    // 智能建筑
  | 'GREEN_BUILDING'    // 绿色建筑
  | 'BIM_APPLICATION'   // BIM应用
  | 'PREFABRICATED'     // 装配式建筑
  | 'RENOVATION';       // 既有建筑改造

// 类别标签映射
export const projectCategoryLabels: Record<ProjectCategory, string> = {
  SMART_BUILDING: '智能建筑',
  GREEN_BUILDING: '绿色建筑',
  BIM_APPLICATION: 'BIM应用',
  PREFABRICATED: '装配式建筑',
  RENOVATION: '既有建筑改造',
};

// 项目状态枚举（后端用数字）
export type ProjectStatus = 0 | 1;  // 0=草稿, 1=已发布

// 状态标签映射
export const projectStatusLabels: Record<ProjectStatus, string> = {
  0: '草稿',
  1: '已发布',
};

// 项目规模
export interface ProjectScale {
  area?: string;
  height?: string;
  investment?: string;
}

// 技术特点
export interface TechnicalFeature {
  title: string;
  description: string;
}

// 项目成果
export interface ProjectAchievement {
  title: string;
  value: string;
  description: string;
}

// 项目列表项响应
export interface ProjectListResponse {
  id: number;
  title: string;
  category: ProjectCategory;
  categoryName: string;
  location: string;
  completionDate: string | null;
  owner: string;
  views: number;
  status: ProjectStatus;
  coverImage: string | null;
  images: string | null;
}

// 项目详情响应
export interface ProjectResponse {
  id: number;
  title: string;
  category: ProjectCategory;
  categoryName: string;
  location: string;
  completionDate: string | null;
  owner: string;
  designer: string;
  contractor: string;
  description: string;
  background: string | null;
  designConcept: string | null;
  views: number;
  status: ProjectStatus;
  coverImage: string | null;
  createdTime: string;
  updatedTime: string;
  // JSON string fields
  highlights: string | null;
  projectAwards: string | null;
  scale: string | null;
  technicalFeatures: string | null;
  achievements: string | null;
  images: string | null;
}

// 创建/更新项目请求
export interface ProjectRequest {
  title: string;
  category: ProjectCategory;
  location?: string;
  completionDate?: string;
  owner?: string;
  designer?: string;
  contractor?: string;
  description?: string;
  background?: string;
  designConcept?: string;
  status?: number;
  coverImage?: string;
  // JSON string fields
  highlights?: string;
  projectAwards?: string;
  scale?: string;
  technicalFeatures?: string;
  achievements?: string;
  images?: string;
}

// 解析后的项目详情（用于前端展示）
export interface ParsedProjectResponse extends Omit<ProjectResponse, 'highlights' | 'projectAwards' | 'scale' | 'technicalFeatures' | 'achievements' | 'images'> {
  highlights: string[];
  projectAwards: string[];
  scale: ProjectScale | null;
  technicalFeatures: TechnicalFeature[];
  achievements: ProjectAchievement[];
  images: string[];
}

// JSON 解析辅助函数
export function parseProjectResponse(project: ProjectResponse): ParsedProjectResponse {
  return {
    ...project,
    highlights: parseJsonArray(project.highlights),
    projectAwards: parseJsonArray(project.projectAwards),
    scale: parseJsonObject<ProjectScale>(project.scale),
    technicalFeatures: parseJsonArray<TechnicalFeature>(project.technicalFeatures),
    achievements: parseJsonArray<ProjectAchievement>(project.achievements),
    images: parseJsonArray(project.images),
  };
}

function parseJsonArray<T = string>(json: string | null): T[] {
  if (!json) return [];
  try {
    return JSON.parse(json);
  } catch {
    return [];
  }
}

function parseJsonObject<T>(json: string | null): T | null {
  if (!json) return null;
  try {
    return JSON.parse(json);
  } catch {
    return null;
  }
}

File: cms/src/main/java/com/assoc/cms/entity/Project.java
package com.assoc.cms.entity;

import com.assoc.common.entity.AuditableEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * Project showcase entity
 */
@Data
@Entity
@Table(name = "cms_project")
@EqualsAndHashCode(callSuper = true)
public class Project extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * Project title
     */
    @Column(name = "title", nullable = false, length = 200)
    private String title;

    /**
     * Project category (legacy enum field, kept for backward compatibility)
     */
    @Enumerated(EnumType.STRING)
    @Column(name = "category", length = 30)
    private ProjectCategory category;

    /**
     * Project category ID (new field for dynamic category management)
     */
    @Column(name = "category_id")
    private Long categoryId;

    /**
     * Project category entity (for fetching category details)
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    private ProjectCategoryEntity categoryEntity;

    /**
     * Location
     */
    @Column(name = "location", length = 200)
    private String location;

    /**
     * Completion date
     */
    @Column(name = "completion_date")
    private LocalDate completionDate;

    /**
     * Owner/Client
     */
    @Column(name = "owner", length = 200)
    private String owner;

    /**
     * Designer
     */
    @Column(name = "designer", length = 200)
    private String designer;

    /**
     * Contractor
     */
    @Column(name = "contractor", length = 200)
    private String contractor;

    /**
     * Project description
     */
    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    /**
     * Highlights (JSON array)
     */
    @Column(name = "highlights", columnDefinition = "TEXT")
    private String highlights;

    /**
     * Awards (JSON array)
     */
    @Column(name = "project_awards", columnDefinition = "TEXT")
    private String projectAwards;

    /**
     * View count
     */
    @Column(name = "views")
    private Integer views = 0;

    /**
     * Background information
     */
    @Column(name = "background", columnDefinition = "TEXT")
    private String background;

    /**
     * Scale information (JSON object: area, height, investment)
     */
    @Column(name = "scale", columnDefinition = "TEXT")
    private String scale;

    /**
     * Design concept
     */
    @Column(name = "design_concept", columnDefinition = "TEXT")
    private String designConcept;

    /**
     * Technical features (JSON array)
     */
    @Column(name = "technical_features", columnDefinition = "TEXT")
    private String technicalFeatures;

    /**
     * Achievements (JSON array)
     */
    @Column(name = "achievements", columnDefinition = "TEXT")
    private String achievements;

    /**
     * Images (JSON array of URLs)
     */
    @Column(name = "images", columnDefinition = "TEXT")
    private String images;

    /**
     * Cover image (Base64 encoded)
     */
    @Column(name = "cover_image", columnDefinition = "TEXT")
    private String coverImage;

    /**
     * Status: 1-published, 0-draft
     */
    @Column(name = "status")
    private Integer status = 0;

    /**
     * Sort order
     */
    @Column(name = "sort_order")
    private Integer sortOrder = 0;

    /**
     * Increment view count
     */
    public void incrementViews() {
        this.views = (this.views == null ? 0 : this.views) + 1;
    }

    /**
     * Check if published
     */
    public boolean isPublished() {
        return this.status != null && this.status == 1;
    }
}
